.github/workflows/build_and_test.yml

```diff
@@ -489,6 +490,8 @@ jobs:
         apt-get update -y
         apt-get install -y ruby ruby-dev
         Rscript -e "install.packages(c('devtools', 'testthat', 'knitr', 'rmarkdown', 'roxygen2'), repos='https://cloud.r-project.org/')"
+        Rscript -e "devtools::install_git('https://github.com/r-lib/pkgdown.git')"
+        Rscript -e "devtools::install_git('https://github.com/amirmasoudabdol/preferably.git')"
```

Comment:
quick question, do we need this for documentation build? we would have to also update https://github.com/apache/spark/blob/master/docs/README.md and https://github.com/apache/spark/blob/master/dev/create-release/spark-rm/Dockerfile#L83-L84 . Can be done separately though.

And can we pin the versions of both libraries instead of using the master branch?

Reply:
> quick question, do we need this for documentation build?

Yes, documentation build invokes pkgdown. 

> And can we pin the versions of both libraries instead of using the master branch?

Of course. We should be even able to pin CRAN version of `pkgdown` (seems like [2.0.x](https://cloud.r-project.org/web/packages/pkgdown/index.html) and [compatible preferably](https://cloud.r-project.org/web/packages/preferably/))) are already published and visible).